File: 3704-count-partitions-with-even-sum-difference/count-partitions-with-even-sum-difference.py
class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        # Brute - Force Solution

        """
        n = len(nums)
        count = 0
        totalSum = []
        for i in range(n - 1):
            sumValue1 = 0
            sumValue2 = 0
            for l in range(0, i + 1):
                sumValue1 += nums[l]
            for r in range(i + 1, n):
                sumValue2 += nums[r]
            totalSum.append(sumValue1 - sumValue2)
        
        for i in totalSum:
            if (i % 2 == 0):
                count += 1

        return count
        """

        # Optimized Solution

        total = sum(nums)
        n = len(nums)
        if (total % 2 != 0):
            return 0
        return n - 1
        

File: 3704-count-partitions-with-even-sum-difference/count-partitions-with-even-sum-difference.java
class Solution {
    public int countPartitions(int[] nums) {
        // Brute - Force Solution

        /*
        int n = nums.length;
        int count = 0;
        int[] totalDiff = new int[n - 1];

        for (int i = 0; i < n - 1; i++) {
            int sumValue1 = 0;
            int sumValue2 = 0;

            for (int l = 0; l <= i; l++) {
                sumValue1 += nums[l];
            }
            for (int r = i + 1; r < n; r++) {
                sumValue2 += nums[r];
            }

            totalDiff[i] = sumValue1 - sumValue2;
        }

        for (int diff : totalDiff) {
            if (diff % 2 == 0) {
                count++;
            }
        }

        return count;
        */

        // Optimized Solution

        int total = 0;
        for (int i : nums)
            total += i;
        if (total % 2 != 0)
            return 0;
        return nums.length - 1;
    }
}
